apps/src/code-studio/components/Libraries/libraryParser.js

```diff
@@ -0,0 +1,80 @@
+import {getFunctionsAndMetadata} from '@cdo/apps/lib/tools/jsinterpreter/JSInterpreter';
+
+/**
+ * Finds all functions in the provided code and returns them with any metadata
+ * @param {string} code The code to parse to get all functions
+ * @returns {array} A list of objects where each object represents a function with its doc comment and params
+ */
+export function getFunctions(code) {
+  return getFunctionsAndMetadata(code);
+}
+
+/**
+ * Takes the input library elements and creates a json string from them to be
+ * uploaded to S3
+ * @param {string} code The code that makes up the library
+ * @param {array} selectedFunctions The list of functions that will be exported from the library
+ * @param {string} libraryName The name of the library to be exported
+ * @returns {string} A JSON string representign the library
+ */
+export function createLibraryJson(code, selectedFunctions, libraryName) {
+  if (
+    typeof code !== 'string' ||
+    !Array.isArray(selectedFunctions) ||
+    typeof libraryName !== 'string' ||
+    !libraryName
+  ) {
+    return;
+  }
+
+  let exportedFunctions = [];
+  let fullConfig = [];
+  for (let selectedFunction of selectedFunctions) {
+    if (!selectedFunction || !selectedFunction.functionName) {
```

Comment:
Nit: is the !selectedFunction boolean redundant with the selectedFunction being created in the forEach?

Reply:
👍